### DISCUSSION TITLE
Welcome to IntexSpaManager Discussions!

### DISCUSSION THREAD

**Discussion by bikemike** [bikemike/IntexSpaManager#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aachtenberg** [comment]
New to this for sure. I see this code is for a d1 mini. What are the pinouts connected to? I have an Intex Purespa greywood deluxe with the remote panel - it stopped working. I'd like to know what you use to connect the d1 mini to the base.

**Reply by bikemike** [reply]
I soldered some wires onto the main PCB board to hook into the display / control unit signals. The base of your model looks similar but if it doesn't have a display connected by a cable then it probably works a little differently. You'd have to open up the base and take a picture of the PCB. I have one somewhere that I have been meaning to upload here with some more details on what I did.

**Reply by aachtenberg** [reply]
![PXL_20210318_113235247 MP](https://user-images.githubusercontent.com/840813/111629125-51781d00-87c7-11eb-81d3-4f8a46d73e55.jpg)
It looks like a LC12S module on the left. That communicates with the remote panel which also has the same chip. I haven't taken apart my remote panel - I am getting it replaced under warranty. Here is a photo from someone who did take apart their remote:

https://photos.google.com/share/AF1QipN-zMlSjcrrOGbNaWrgr3r414NnC-93LxP9-5h0T4fBRo_DrBB67R1GKKzsZI2S2w?key=QzZfcDlBMGR1cTlJUmxVcl9EbFBkeVVDMm5ud0tB

**Reply by bikemike** [reply]
It looks quite different than mine so what I did will not work for you. You could probably figure out the communication protocol by putting a logic analyzer on the tx/rx of the LC12S.  Then you could make your own "LC12S / ESP8266" device that would allow you to control and monitor over wifi. Do you have any experience with this type of work?

If you're getting a new controller you may be able to use the LC12S from the broken controller. It would be interesting to see if multiple controllers can work at the same time.

**Reply by aachtenberg** [reply]
Little experience with HW like this outside an academic environment (university). Never had an interest into looking into Ardunio until now. There is a video posted on youtube where a person said they used a standalone wifi to RF bridge but I'mt not sure how they would be able to do that if chip operates at 2.4Ghz.  You can tell on the video comments he is not very responsive to questions!
https://www.youtube.com/watch?v=5Lx3TuCBxhk&ab_channel=maultrommel

Could you recommend a logic analyzer? I'm seriously considering creating a LC12S/ESP8266 device.

**Reply by bikemike** [reply]
It's really too bad that guy didn't share his work.

You can get cheap saleae clones from aliexpress (~$10). But you may not need a logic analyzer. If you hook up to two pins on an ESP8266 you can set them up as software serial rx pins and listen in. You just need to get the baud correct. Maybe it's 9600? Then output the results to a serial console. I have some code that does this that I used for another project. I'll upload it.

**Reply by bikemike** [reply]
I've uploaded the serial monitoring code here:

https://github.com/bikemike/SerialTxRxMonitor

**Reply by aachtenberg** [reply]
Thanks! I've cloned the repo. I'll have a ESP8266 here this week so I'll get started after that.

**Reply by aachtenberg** [reply]
Electronics are starting to arrive. I managed to find a thread on this subject on a community forum on jeedom. There is a user that appears to have done what I am trying to do. One of his comments:
![image](https://user-images.githubusercontent.com/840813/112228794-3e6eae00-8c08-11eb-80b4-b4a82824d1a3.png)

**Reply by aachtenberg** [reply]
I'm conversing with him so hopefully I can get some code.

**Reply by bikemike** [reply]
Hey that's cool. That should save you a lot of time. That guy should get his code up on github too!

ESP8266 should work just fine. You can use software serial for other pins as my SerialTxRxMonitor does.

**Reply by aachtenberg** [reply]
I figured it would be just fine but I thought I'd share just in case.

**Reply by Yogui79** [reply]
> Hey that's cool. That should save you a lot of time. That guy should get his code up on github too!
> 
> ESP8266 should work just fine. You can use software serial for other pins as my SerialTxRxMonitor does.

Hello 

I'm the guy from the jeedom forum.

software serial doesn't work fine on ESP8266 according my experiance I have a lot off reboot from the ESP.

I will share my work regarding the control from the spa with an ESP32 Over mqtt in a few day

**Reply by aachtenberg** [reply]
That's great. My LC12s chips are still in transit from Aliexpress. Hopefully arriving in the next week or so. I do have a couple of ESP32 chips too.

**Reply by Yogui79** [reply]
Hello 
I just release my  code
https://github.com/Yogui79/IntexPureSpa

**Reply by aachtenberg** [reply]
Looking forward to getting your code working - LC12s arrive early next week!

**Reply by Yogui79** [reply]
> Looking forward to getting your code working - LC12s arrive early next week!

Hello

Did you receive your Lc12s

We have detected that the network id is not depending spa model but spa please open a discussion when you receive your lc12 on my git I will explain you how to handle that .
I Will also try to update my git with the information asap
